### ISSUE TITLE
Requires now an MPI with Fortran support

### ISSUE THREAD

**Issue by bertwesarg** [Parallel-NetCDF/PnetCDF#68]
Since 01857cc13d70ef49164b1ab19228d2937255a311 an MPI with Fortran support is required, because the checks for the MPI Fortran datatypes where removed and thus made mandatory. The commit message only talks about MPI Version but not Fortran support, thus I assume this was not intended.

**Comment by wkliao** [comment]
PnetCDF's Fortran feature is enabled by default, but can be disabled at configure
time by using `--disable-fortran`.

Because MPI standard revises both C and Fortran APIs and constants simultaneously,
datatypes in C and Fortran are consistent. Isn't checking MPI version sufficient?

**Comment by bertwesarg** [comment]
Sorry, I should have stated this, but this was already build with `--disable-fortran`. But you also see, that these checks were normally protected with `if test "x${has_fortran}" = xyes ; then`.

This was also an Open MPI 4.0.5 build with `--disable-mpi-fortran` and mpi.h only defines `MPI_INTEGER1` if supported:

```c
#if OMPI_HAVE_FORTRAN_INTEGER1
#define MPI_INTEGER1 OMPI_PREDEFINED_GLOBAL(MPI_Datatype, ompi_mpi_integer1)
#endif
```

**Comment by wkliao** [comment]
I may not understand your point correctly.
Are you suggesting that when `--disable-fortran` is used, PnetCDF's
configure.ac should still check whether MPI Fortran data types, such
as MPI_INTEGER1, are defined?

**Comment by wkliao** [comment]
I think I now understand what you are saying.
When OpenMPI is built with Fortran feature disabled, i.e. using --disable-mpi-fortran
in the configure command line, all predefined MPI Fortran datatypes, such as
MPI_INTEGER1, will not be defined in the MPI header file, mpi.h. However, PnetCDF
uses those Fortran predefined datatypes anyway without checking whether they
are defined in mpi.h. An error occurs when compiling file 
`src/drivers/common/dtype_decode.c` with error messages such as below.
```
../../../../PnetCDF/src/drivers/common/dtype_decode.c:50:17: error: ‘MPI_INTEGER1’ undeclared (first use in this function); did you mean ‘MPI_INTEGER’?
     if (type == MPI_INTEGER1)
                 ^~~~~~~~~~~~
                 MPI_INTEGER
```
Thanks for reporting this issue. I will soon develop a patch to fix it.

**Comment by bertwesarg** [comment]
Correct, thanks for taking care.

**Comment by wkliao** [comment]
fixed in #69
